native/src/main/java/io/ballerina/lib/redis/connection/RedisConnectionManager.java

```diff
@@ -396,4 +392,26 @@ private SslOptions constructSslOptions(SecureSocket secureSocket) {
 
         return sslOptionsBuilder.build();
     }
+
+    /**
+     * Test the pooled connection by borrowing and returning a connection. This is to foresee any connection issues,
+     * when pooling is enabled.
+     *
+     * @throws RedisConnectorException if an error occurs while borrowing or returning a connection
+     */
+    private void testConnectionPool() throws RedisConnectorException {
+        StatefulConnection<K, V> connection = null;
+        try {
+            connection = objectPool.borrowObject();
+            if (connection == null) {
+                throw new RedisConnectorException("Failed to borrow a connection from the pool");
+            }
+        } catch (Exception e) {
+            throw new RedisConnectorException(e.getMessage());
```

Comment:
Here I think we should preserve the underlying exception (e) as the cause. I think that information would be valuable when we check the error in the ballerina side.